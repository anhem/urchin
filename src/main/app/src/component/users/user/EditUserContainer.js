import React, {Component} from "react";
import {connect} from "react-redux";
import EditUser from "./EditUser";
import {addGroup, deleteUser, getGroupsForUser, getUser, removeGroup, setUser} from "../../../action/userAction";
import {getGroups} from "../../../action/groupAction";

export class EditUserContainer extends Component {

    componentWillMount() {
        let userId = parseInt(this.props.match.params.id, 10);
        this.props.getUser(userId);
        this.props.getGroupsForUser(userId);
        this.props.getGroups();
    }

    filterOutAvailableGroups() {
        let groupIds = this.props.groupsForUser.map(group => group.groupId);

        return this.props.groups.filter(group =>
            groupIds.indexOf(group.groupId) === -1
        );
    }

    render() {
        let availableGroups = this.filterOutAvailableGroups();

        return (
            <EditUser
                user={this.props.user}
                groupsForUser={this.props.groupsForUser}
                availableGroups={availableGroups}
                callbacks={{
                    addGroup: this.props.addGroup,
                    removeGroup: this.props.removeGroup,
                    deleteUser: this.props.deleteUser,
                    setUser: this.props.setUser
                }}
            />
        )
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        getUser: (userId) => {
            dispatch(getUser(userId))
        },
        setUser: (user) => {
            dispatch(setUser(user))
        },
        deleteUser: (userId) => {
            dispatch(deleteUser(userId))
        },
        getGroupsForUser: (userId) => {
            dispatch(getGroupsForUser(userId))
        },
        getGroups: () => {
            dispatch(getGroups())
        },
        addGroup: (userId, groupId) => {
            dispatch(addGroup(userId, groupId))
        },
        removeGroup: (userId, groupId) => {
            dispatch(removeGroup(userId, groupId))
        }
    }
};

const mapStateToProps = (state) => {
    return {
        user: state.userData.user || {},
        groupsForUser: state.userData.groupsForUser || [],
        groups: state.groupData.groups || []
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(EditUserContainer)