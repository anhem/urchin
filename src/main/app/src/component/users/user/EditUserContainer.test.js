import React from 'react';
import {mount, shallow} from 'enzyme'
import toJson from 'enzyme-to-json';
import {EditUserContainer} from "./EditUserContainer";

describe('EditUserContainer', () => {

    let props = {
        match: {
            params: {
                id: 0
            }
        },
        user: {},
        groupsForUser: [],
        groups: [],
        getUser: jest.fn(),
        getGroupsForUser: jest.fn(),
        getGroups: jest.fn(),
    };

    it('renders without crashing', () => {
        expect(mount(<EditUserContainer {...props}/>).length).toEqual(1);
    });

    it('match snapshot', () => {
        expect(toJson(shallow(<EditUserContainer {...props}/>))).toMatchSnapshot();
    });
});
