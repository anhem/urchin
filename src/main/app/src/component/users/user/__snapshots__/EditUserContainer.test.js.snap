// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditUserContainer match snapshot 1`] = `
<EditUser
  availableGroups={Array []}
  callbacks={
    Object {
      "addGroup": undefined,
      "deleteUser": undefined,
      "removeGroup": undefined,
      "setUser": undefined,
    }
  }
  groupsForUser={Array []}
  user={Object {}}
/>
`;
